src/Cli/dotnet/Commands/Tool/Execute/ToolExecuteCommand.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using Microsoft.DotNet.Cli.CommandFactory;
+using Microsoft.DotNet.Cli.CommandFactory.CommandResolution;
+using Microsoft.DotNet.Cli.Commands.Tool.Install;
+using Microsoft.DotNet.Cli.ToolPackage;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Cli.Commands.Tool.Execute
+{
+    internal class ToolExecuteCommand(ParseResult result) : CommandBase(result)
+    {
+        private readonly PackageIdentity? _packageToolIdentityArgument = result.GetValue(ToolExecuteCommandParser.PackageIdentityArgument);
+        private readonly IEnumerable<string> _forwardArguments = result.GetValue(ToolExecuteCommandParser.CommandArgument);
+        private readonly bool _allowRollForward = result.GetValue(ToolExecuteCommandParser.RollForwardOption);
+        private readonly string _configFile = result.GetValue(ToolExecuteCommandParser.ConfigOption);
+        private readonly string[] _sources = result.GetValue(ToolExecuteCommandParser.SourceOption) ?? [];
+        private readonly string[] _addSource = result.GetValue(ToolExecuteCommandParser.AddSourceOption) ?? [];
+        private readonly bool _ignoreFailedSources = result.GetValue(ToolCommandRestorePassThroughOptions.IgnoreFailedSourcesOption);
+        private readonly bool _interactive = result.GetValue(ToolExecuteCommandParser.InteractiveOption);
+        private readonly VerbosityOptions _verbosity = result.GetValue(ToolExecuteCommandParser.VerbosityOption);
+        private readonly bool _yes = result.GetValue(ToolExecuteCommandParser.YesOption);
+        private readonly bool _prerelease = result.GetValue(ToolExecuteCommandParser.PrereleaseOption);
+
+        public override int Execute()
+        {
+            if (!UserAgreedToRunFromSource() || _packageToolIdentityArgument is null)
+            {
+                return 1;
+            }
+
+            if (_allowRollForward)
+            {
+                _forwardArguments.Append("--allow-roll-forward");
+            }
+
+            PackageId packageId = new PackageId(_packageToolIdentityArgument.Id);
+
+            VersionRange versionRange = _parseResult.GetVersionRange();
+
+            string tempDirectory = PathUtilities.CreateTempSubdirectory();
```

Comment:
I may just not understand the feature properly, but I thought marcpopMSFT told me that the plan was to execute the tool directly from the nuget cache? Though it would be pretty fast as long as it'd already been downloaded.